src/split.c

```diff
@@ -115,7 +115,7 @@ rcutils_split(
   if (rcutils_string_array_fini(string_array) != RCUTILS_RET_OK) {
     error_msg = rcutils_format_string(allocator, "FATAL: %s. Leaking memory", error_msg);
   }
-  RCUTILS_SET_ERROR_MSG(error_msg, allocator);
+  rcutils_set_formatted_error(error_msg, allocator);
```

Comment:
This is a memory leak and should be fixed, but again, I don't think a special function is required here. Also this change will change the file and line number associated with this error state.